sqlglot/parser.py

```diff
@@ -5339,7 +5350,11 @@ def _parse_alter(self) -> exp.AlterTable | exp.Command:
 
         parser = self.ALTER_PARSERS.get(self._prev.text.upper()) if self._prev else None
         if parser:
-            actions = ensure_list(parser(self))
+            parse_result = parser(self)
+            if parse_result is None:
+                return self._parse_as_command(start)
+
+            actions = ensure_list(parse_result)
```

Comment:
This can be reverted, if `parse_result` is `None` then `ensure_list` will return an empty list for `actions` and the `self.expression` call below will fail since `actions` is a required argument.

```suggestion
            actions = ensure_list(parse_result)
```

Reply:
I think this would be considered a breaking change though since the output of this would then always be an exp.AlterTable instead of exp.Command. Would it be better to change it to 
```
            if not self._curr and actions is not None:
```
then?

Reply:
Nvm, I was wrong apparently. Your change makes sense after all because failing at `expression` is no good.

Reply:
> I think this would be considered a breaking change though since the output of this would then always be an exp.AlterTable instead of exp.Command. Would it be better to change it to
> 
> ```
>             if not self._curr and actions is not None:
> ```
> 
> then?

Yeah, you can do `and actions` because `actions` will be an empty list. Can you make this change as well?